stablehlo/reference/Ops.cpp

```diff
@@ -323,6 +335,36 @@ Tensor evalAndOp(const Tensor &lhs, const Tensor &rhs, TensorType resultType) {
   return result;
 }
 
+Tensor evalBatchNormInferenceOp(const Tensor &operand, const Tensor &scale,
+                                const Tensor &offset, const Tensor &mean,
+                                const Tensor &variance, APFloat epsilon,
+                                Axis featureIndex, TensorType resultType) {
+  auto scaleBroadcast =
+      evalBroadcastInDimOp(scale, {featureIndex}, operand.getType());
+  auto offsetBroadcast =
+      evalBroadcastInDimOp(offset, {featureIndex}, operand.getType());
+  auto meanBroadcast =
+      evalBroadcastInDimOp(mean, {featureIndex}, operand.getType());
+  auto varianceBroadcast =
+      evalBroadcastInDimOp(variance, {featureIndex}, operand.getType());
+  auto epsilonBroadcast = evalBroadcastInDimOp(
+      makeTensor(DenseElementsAttr::get(
+          RankedTensorType::get({}, operand.getElementType()), {epsilon})),
```

Comment:
Nice catch! Let's change the spec to say `constant(epsilon, type(operand))` instead of just `constant(epsilon)`?